cardano-cli/src/Cardano/CLI/EraBased/Run/Transaction.hs

```diff
@@ -610,9 +609,7 @@ runTxBuild
 
       return balancedTxBody
 
-    (CardanoMode, LegacyByronEra) -> left TxCmdByronEra
```

Comment:
This case handler for `Byron` always fails so our legacy `transaction build` could never have worked on Byron.

Reply:
So `CardanoMode` here captures everything `> ShelleyMode` right?

Reply:
`CardanoMode` is the one that supports all the eras.